components/ShareButton.js

```diff
@@ -0,0 +1,45 @@
+import styled from "styled-components";
+
+export default function ShareButton({ emotion }) {
+  async function handleShare() {
+    const shareData = {
+      title: "Emotion Card",
+      text: `I'm feeling ${emotion.type.name} with intensity ${
+        emotion.intensity
+      }. Note: ${emotion.notes || ""}`,
+      url: window.location.href,
+    };
+
+    try {
+      if (navigator.share) {
+        await navigator.share(shareData);
```

Comment:
since you are not doing anything else after the navigator.share, you don't need to await the Promise.